lib/AST/DeclContext.cpp

```diff
@@ -804,10 +804,12 @@ ArrayRef<Decl *> IterableDeclContext::getParsedMembers() const {
 
 ArrayRef<Decl *> IterableDeclContext::getSemanticMembers() const {
   ASTContext &ctx = getASTContext();
-  return evaluateOrDefault(
+  auto result = evaluateOrDefault(
       ctx.evaluator,
       SemanticMembersRequest{const_cast<IterableDeclContext *>(this)},
       ArrayRef<Decl *>());
+  assert(result.size() <= getMemberCount());
```

Comment:
This assertion might deserve a comment: When would there ever be fewer semantic members than members? Are there noop members?